samples/bluetooth/mtu_update/central/src/central_mtu_update.c

```diff
@@ -82,7 +82,8 @@ static uint8_t discover_func(struct bt_conn *conn, const struct bt_gatt_attr *at
 			printk("Discover failed (err %d)\n", err);
 		}
 	} else if (!bt_uuid_cmp(discover_params.uuid, BT_UUID_MTU_TEST_NOTIFY)) {
-		memcpy(&uuid, BT_UUID_GATT_CCC, sizeof(uuid));
+		memset(&uuid, 0, sizeof(uuid));
+		memcpy(&uuid, BT_UUID_GATT_CCC, sizeof(struct bt_uuid_16));
 		discover_params.uuid = &uuid.uuid;
```

Comment:
It should also be possible to do 
```
static const struct bt_uuid *ccc_uuid = BT_UUID_GATT_CCC;

discover_params.uuid = ccc_uuid;
```

As an alternative (that's how we often do it in the LE Audio implementation)

Reply:
but muh memory!
I'll align it with yours then :)

Reply:
Out of scope of course: Would it work to have a c-file with `const struct bt_uuid ccc_uuid`, etc..? With an accompanying header with `extern const struct bt_uuid ccc_uuid`. The linker will strip out unused symbols, right? I think this will actually save ROM and RAM.

Reply:
> EDIT: Oh. It's a bit more complicated, since bt_uuid is a "superclass", so it must be a pointer.

Yeah, the `bt_uuid` is not that easy to work with :D That type has also caused countless coverity false positives due to how it's implemented

Reply:
I though trough it again, and removed the EDIT. We should just have the real types on the objects (e.g. bt_uuid_16). And maybe define a macro for doing the CONAINTER_OF-inverse `&ccc_uuid.uuid`.

Reply:
> I though trough it again, and removed the EDIT. We should just have the real types on the objects (e.g. bt_uuid_16). And maybe define a macro for doing the CONAINTER_OF-inverse `&ccc_uuid.uuid`.

That would certainly make it easier to work with. I am not familiar with the design decisions on why it is how it is now, but I assume it's a form of optimization.